lib/autobahn/consumer.rb

```diff
@@ -59,6 +59,13 @@ def unsubscribe!(timeout=30)
             subscription.cancel
           end
           @subscriptions = nil
+          @deliver.set(false)
```

Comment:
I don't remember the Autobahn internals properly, so I don't remember if this deals with batches of messages or not. If the `@deliver` flag is checked between delivery of messages in the same batch, we definitely do not want to set it to false here. As is, the deliver method stops immediately when `@deliver` is cleared and doesn't care if there are further messages enqueued. If these are full messages this should be ok, as RabbitMQ is supposed to handle that. If they are part of batches, we might process half batches and "exit cleanly" with this change.